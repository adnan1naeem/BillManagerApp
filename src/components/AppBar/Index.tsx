import Appbar from  "./AppBar.component";
export default Appbar; 