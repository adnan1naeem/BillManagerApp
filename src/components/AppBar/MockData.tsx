
export const Data = {

  backbuttonTrue: true,

  backbuttonfalse: false,
  title: "BillManager",
  style: { color: 'yellow' }

}