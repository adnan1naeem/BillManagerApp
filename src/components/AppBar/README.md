#  AppBar Component

## Compatibility

| 🌏 Web                   | 🖥 Electron              | 📱 React Native    |
| :----------------------: | :----------------------: | :----------------: |
| :heavy_multiplication_x: | :heavy_multiplication_x: | :heavy_check_mark: |

## Screenshots

| 🌏 Web | 🖥 Electron | 📱 React Native                     |
| :----: | :---------: | :---------------------------------: |
| TBD    | TBD         | ![AppBar](./Screenshots/Appbar.png) |
|        |
| TBD    | TBD         | ![AppBar](./Screenshots/AppBar.png) |

## Props

| Name       | Type    | Default | Description                                |
| :--------- | :------ | :------ | :----------------------------------------- |
| title      | string  |         | Adds title of Appbar                       |
| styles     | object  |         | It is used for styling Appbar              |
| BackButton | boolean |         | It is used for adding BackButton in Appbar |


## How to use

```react
import React from 'react';
 <AppBar
  title="BillManager"
  BackButton={true}
 />
```

## Tests
