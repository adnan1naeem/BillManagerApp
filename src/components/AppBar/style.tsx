export const styles: any = {

  View: {
    alignItems: 'center',
    height: 56,
    display: 'flex',
    flexDirection: 'row',
    flex: 1,
    backgroundColor: 'blue'
  },
  buttonColor: {
    color: 'white',
    flex: 1
  },
  titlecolor: {
    color: 'white',
    marginLeft: 10

  }
}