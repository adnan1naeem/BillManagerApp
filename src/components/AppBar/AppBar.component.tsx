import { View, Text } from 'react-native'
import React from 'react'
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import { Link } from 'react-router-native'
import { styles } from "./style"

export interface Props {
  /**
	 * title of the AppBar 
	 * 
	 */

  title: string,

  /**
	 * BackButton hide and show prop
	 * 
	 */
  BackButton?: boolean,

  /**
	 * Adding styles in the AppBar
	 * 
	 */
  style?: object,

}

const AppBar: React.StatelessComponent<Props> = (props) => {
  const { title } = props;

  return (
    <View style={styles.View}>
      {
        props.BackButton ?
          <Link
            to={'/'}
          >
            <MaterialCommunityIcons
              name="arrow-left"
              size={25}
              color="white"
            />
          </Link>
          : null
      }
      <Text style={styles.titlecolor}>{title}</Text>
    </View>
  );
}
export default AppBar;


