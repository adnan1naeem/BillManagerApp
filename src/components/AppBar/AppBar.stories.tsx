import React from 'react';
import { storiesOf } from '@storybook/react-native';
import AppBar from './AppBar.component';
import { withKnobs, boolean, text } from '@storybook/addon-knobs/react';
import { View, StyleSheet } from 'react-native';
import { ScrollView, Button } from "react-native";
import { Data } from "./MockData"
import { withInfo } from '@storybook/addon-info';
import { LinkTo, linkTo } from '@storybook/addon-links';




storiesOf('Appbar', module)
  .addDecorator(getStory => <View>{getStory()}</View>)
  .addDecorator(withKnobs)

  .add(' back button Appbar story without knobs simple', () => (

    <ScrollView>
      <AppBar
        title="BillManager"
        BackButton={boolean('backbutton', Data.backbuttonTrue)}
        to={'/'}
      />
    </ScrollView>
  ))

  .add('back button Appbar story without knobs simple', () => (

    <ScrollView>
      <AppBar
        title="BillManager"
        BackButton={boolean('backbutton', Data.backbuttonTrue)}
      />
    </ScrollView>
  ))

  .add(' back button Appbar story with knobs', () => (

    <ScrollView>
      <AppBar
        title={text('title', 'BillManager')}
        BackButton={boolean('backbutton', Data.backbuttonTrue)}
      />
    </ScrollView>
  ))

  .add('without  back button Appbar story', () => (

    <ScrollView>
      <AppBar
        title={text('title', 'BillManager')}
        BackButton={boolean('backbutton', Data.backbuttonfalse)}
        style={Data.style}
      />
    </ScrollView>
  ))

  .add('AppBar title knob story', () => (

    <ScrollView>
      <AppBar
        title={text('title', 'BillManager')}
        BackButton={boolean('backbutton', Data.backbuttonfalse)}
        style={Data.style}
      />
    </ScrollView>
  ))






