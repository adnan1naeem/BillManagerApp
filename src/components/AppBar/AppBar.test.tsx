import * as React from 'react'
import { shallow, configure, mount } from 'enzyme';
import AppBar from "./AppBar.component";
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';


configure({ adapter: new Adapter() })

test('renders correctly', () => {
  shallow(<AppBar title="ssss" />);
});


test('renders correctly', () => {
  const tree = renderer.create(<AppBar title="ssss" />).toJSON();
  expect(tree).toMatchSnapshot();
});



