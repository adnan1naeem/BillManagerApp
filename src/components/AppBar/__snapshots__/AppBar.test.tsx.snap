// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "blue",
      "display": "flex",
      "flex": 1,
      "flexDirection": "row",
      "height": 56,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
      }
    }
  >
    ssss
  </Text>
</View>
`;
